lib/matplotlib/figure.py

```diff
@@ -362,7 +362,7 @@ def _repr_html_(self):
         # We can't use "isinstance" here, because then we'd end up importing
         # webagg unconditiionally.
         if (self.canvas is not None and
-            'WebAgg' in self.canvas.__class__.__name__):
+                'WebAgg' in self.canvas.__class__.__name__):
```

Comment:
Actually the whole condition should just get rewritten as `if type(self.canvas).__name__ == "FigureCanvasWebAgg": ...`